rfd/0239-windows-ca-split.md

```diff
@@ -0,0 +1,166 @@
+---
+authors: Alan Parra (alan.parra@goteleport.com)
+state: draft
+---
+
+# RFD 0239 - Windows CA split
+
+## Required Approvers
+
+* Engineering: @rosstimothy && @zmb3
+* Security: (@rob-picard-teleport || @klizhentas)
+* Product: @klizhentas
+
+## What
+
+Split the "windows" (aka Desktop Access) CA from the User CA.
+
+## Why
+
+The "windows" CA is used to generate user certificates for Desktop Access. Those
+certificates are used to perform smart card authentication on RDP (Remote
+Desktop Protocol).
+
+Currently the User CA issues those certificates, as they are end-user certs.
+Splitting the CAs improves Teleport's security posture by introducing a more
+specialized CA (the "windows" CA needs to be trusted by downstream Windows
+systems) and lets both CAs be rotated independently.
+
+This is a task we've been long meaning to address (see [#10111](
+https://github.com/gravitational/teleport/issues/10111)).
+
+## UX
+
+The CA split borrows its methodology and UX from the previous [db and db_client
+split][docs-db-ca-migration].
+
+The new CA is called "windows", as that term is already used in the product's UX
+(for example, `/webapi/auth/export?type=windows` and `tctl auth export
+--type=windows`).
+
+Honorable mention to [RFD 0168][rfd0168].
+
+[docs-db-ca-migration]: https://goteleport.com/docs/zero-trust-access/management/operations/db-ca-migrations/#teleport-db_client-ca-migration
+[rfd0168]: https://github.com/gravitational/teleport/blob/master/rfd/0168-database-ca-split.md
+
+### New Teleport cluster
+
+New Teleport clusters are created with distinct User and Windows CAs. No further
+action is needed from the user.
+
+### Existing Teleport cluster
+
+Existing Teleport clusters have the User CA copied into Windows CA on startup. A
+CA rotation is required to create distinct key material.
+
+Neither User or Windows CA is rotated automatically, nor is there any forced
+rotation relationship between them. This follows our [database CA
+migration][docs-db-ca-migration] advice.
+
+It is recommended to allow a safety evaluation period of the new release before
+User or Windows CA rotations, to avoid potential complications due to version
+rollbacks.
+
+## Details
+
+The Windows CA, if it doesn't exist, is created during Teleport initialization:
+
+1. If the UserCA exists: as a copy of the UserCA (existing cluster)
+    * Implemented after [init migrations][init-migrations]
```

Comment:
We've intentionally avoided any new automatic backend database migrations since they tend to cause nothing but problems and we need to support Auth servers running different versions for extended periods of time.

Since this requires a rotation, can we instead of automagically creating a copy of the UserCA on start up, do the copy as part of the rotation?

Reply:
> If we rotate UserCA then what happens? They split and both rotate?

Hrm I hadn't considered that case. What would happen if we only rotated the UserCA and left the WindowsCA as non-existent?

Reply:
> Since this requires a rotation

First of all let's just be honest and say that most people are never going to do the rotation, so we'll have to fall back to reading the UserCA if the WindowsCA doesn't exist forever